pkg/metrics/serve.go

```diff
@@ -110,13 +117,39 @@ func (s *Server) CollectFsMetrics(ctx context.Context) {
 			})
 		}
 	}
-	s.fsCollector.MetricsVec = fsMetricsVec
-	s.fsCollector.Collect()
+	if fsMetricsVec != nil {
+		s.fsCollector.MetricsVec = fsMetricsVec
+		s.fsCollector.Collect()
+	}
+}
+
+func (s *Server) CollectInflightMetrics(ctx context.Context) {
+	// Collect inflight metrics from daemons.
+	daemons := s.pm.ListDaemons()
+	var inflightMetricsVec []*types.InflightMetrics
+	for _, d := range daemons {
+		inflightMetrics, err := d.GetInflightMetrics()
+		if err != nil {
+			log.G(ctx).Errorf("failed to get inflight metric: %v", err)
+			continue
+		}
+		inflightMetricsVec = append(inflightMetricsVec, inflightMetrics)
+	}
+	if inflightMetricsVec != nil {
+		s.inflightCollector.MetricsVec = inflightMetricsVec
+		s.inflightCollector.Collect()
+	}
 }
 
 func (s *Server) StartCollectMetrics(ctx context.Context) error {
 	// TODO(renzhen): make collect interval time configurable
 	timer := time.NewTicker(time.Duration(1) * time.Minute)
+	// The timer period is same as the period for determining hang IOs.
```

Comment:
Grammar mistakes, it should be `is the same`.

Reply:
Fixed.  😊